bot/code_review_bot/revisions.py

```diff
@@ -269,13 +269,26 @@ def analyze_patch(self):
         assert isinstance(self.patch, str), "Invalid patch type"
 
         # List all modified lines from current revision changes
-        patch = Patch.parse_patch(self.patch, skip_comments=False)
-        assert patch != {}, "Empty patch"
+        # associated with their diffs
+        # rs-parsepatch guarantees the order is the same in both lists
+        patch_stats = zip(
+            rs_parsepatch.get_lines(self.patch), rs_parsepatch.get_diffs(self.patch),
```

Comment:
This is... super painful. We have to parse the patch twice this way.

Reply:
I fixed the local patch, and expanded it to make it more readable. The code is way simpler now